### ISSUE TITLE
Error: Cannot read properties of undefined (reading 'flatMap')

### ISSUE THREAD

**Issue by naten3** [serverless-stack/sst#2570]
We're getting an error in CI/CD from [this](
https://github.com/serverless-stack/sst/blob/3207e54b8d5da51332096c0c2646f304249e4301/packages/sst/src/runtime/handlers/node.ts#L228) piece of code. Unfortunately I can't reproduce it locally so I don't have additional information about what the original exception was. Can we have some fallback behavior for when the exception doesn't have an `errors` property? Apparently it's possible.

**Comment by jayair** [comment]
What's the error in your CI/CD logs?

**Comment by naten3** [comment]
```
Trace: TypeError: Cannot read properties of undefined (reading 'flatMap')
    at Object.build (file:///opt/atlassian/pipelines/agent/build/node_modules/sst/runtime/handlers/node.js:186:43)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async task (file:///opt/atlassian/pipelines/agent/build/node_modules/sst/runtime/handlers.js:34:31)
    at async Object.build (file:///opt/atlassian/pipelines/agent/build/node_modules/sst/runtime/handlers.js:78:23)
    at async file:///opt/atlassian/pipelines/agent/build/node_modules/sst/constructs/Function.js:193:32
    at process.<anonymous> (file:///opt/atlassian/pipelines/agent/build/node_modules/sst/cli/sst.js:58:17)
    at process.emit (node:events:525:35)
    at process.emit (node:domain:489:12)
    at process._fatalException (node:internal/process/execution:149:25)
    at processPromiseRejections (node:internal/process/promises:279:13)
    at processTicksAndRejections (node:internal/process/task_queues:97:32)
```

**Comment by thdxr** [comment]
Juse released #2606 which should at least properly print out this error - can you give it a try?

**Comment by naten3** [comment]
Actually we ran into #2587 immediately after getting this error. And now that #2587 is fixed I can't reproduce this error. But I appreciate the logging fix in case we see it again. Thank you!
